packages/db/src/db.ts

```diff
@@ -45,26 +53,62 @@ export class TruffleDB {
     return await execute(this.schema, document, null, this.context, variables);
   }
 
-  async loadCompilations(result: WorkflowCompileResult) {
-    const saga = generateCompileLoad(result, {
+  async getWorkspaceResponse(generatorRequest: WorkspaceRequest) {
+    const { request, variables }: WorkspaceRequest = generatorRequest;
+
+    const response: WorkspaceResponse = await this.query(request, variables);
+
+    return response;
+  }
+  
+  async loadNames(
+    project: DataModel.IProject,
+    contractsByCompilation: Array<DataModel.IContract[]>
+  ) {
```

Comment:
I'm not keen on the arguments to this method here; can we figure out something cleaner?